src/wasm/literal.cpp

```diff
@@ -37,6 +37,10 @@ Literal::Literal(Type type) : type(type) {
     assert(type != Type::unreachable && (!type.isRef() || type.isNullable()));
     if (type.isException()) {
       new (&exn) std::unique_ptr<ExceptionPackage>();
+    } else if (isGCData()) {
+      new (&gcData) std::shared_ptr<GCData>();
+    } else if (type.isRtt()) {
+      new (&rttSupers) auto(std::make_unique<RttSupers>());
```

Comment:
```suggestion
      new (&rttSupers) std::unique_ptr<RttSupers>();
```

Reply:
As you both made the same comment, it seems like either

1. I am wrong, and your suggestion is valid.
2. The difference between the two in C++ is really not clear enough.

As I was starting to doubt myself I built your suggestion and it crashes as I'd expect, so I think it's option 2, or else something else is wrong...